### PR TITLE
Feature/venue org archives

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/81510

Support Venue and Organizer archives:
* extend `Tribe__Duplicate__Post` to support searches
* tweak and extend matching strategies
* avoid 10000 JOIN queries splitting queries

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | A cast statement must be followed by a single space | [Post.php:226](https://github.com/moderntribe/tribe-common/blob/24886dc77f3521bf41f01e60573e34e2cc838332/src/Tribe/Duplicate/Post.php#L226)
:x: | A cast statement must be followed by a single space | [Post.php:226](https://github.com/moderntribe/tribe-common/blob/24886dc77f3521bf41f01e60573e34e2cc838332/src/Tribe/Duplicate/Post.php#L226)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Cache.php [hunk 1]
```diff
@@ -122,5 +134,94 @@ public function set_last_occurrence( $action, $timestamp = 0 ) {
...
+				}
+			}
+			$key .= maybe_serialize( $component );
+		}
+
+		return $this->get_id( $prefix . md5( $key ) );
+	}
+
+	/**
+	 * Whether a offset exists
```
Comment (borkweb) [inline src/Tribe/Cache.php]: `an offset`

File: src/Tribe/Cache.php [hunk 2]
```diff
@@ -122,5 +134,94 @@ public function set_last_occurrence( $action, $timestamp = 0 ) {
...
+	 *
+	 * @link  http://php.net/manual/en/arrayaccess.offsetexists.php
+	 * @param mixed $offset <p>
+	 *                      An offset to check for.
+	 *                      </p>
```
Comment (borkweb) [inline src/Tribe/Cache.php]: Are the `<p>` tags in this docblock intentional?

Comment (borkweb) [inline src/Tribe/Cache.php]: They are in other docblocks, too. They _seem_ accidental.

File: src/Tribe/Duplicate/Post.php [hunk 3]
```diff
@@ -152,35 +313,98 @@ public function find_for( array $postarr ) {
...
+			while ( count( $join ) ) {
+				$current_wheres = array_splice( $custom_fields_where_frags, 0, $join_limit );
+				$current_joins = array_splice( $join, 0, $join_limit );
+
+				$this_join = implode( "\n", $current_joins );
+
+				$this_where = "\n" . implode( " \n{$this->where_operator} ", array_merge( $where_frags, $current_wheres ) );
+				$this_where = sprintf( '%s AND (%s)', $post_type_conditional, $this_where );
+
+				$queries[] = "SELECT DISTINCT {$wpdb->posts}.ID from {$wpdb->posts} {$this_join} \nWHERE {$this_where}";
```
Comment (borkweb) [inline src/Tribe/Duplicate/Post.php]: Nice and clever!

File: src/Tribe/Cache.php (hunk 2 above)
Comment (lucatume) [inline src/Tribe/Cache.php]: Straight from PHP documentation.

Comment (lucatume) [inline src/Tribe/Cache.php]: But fixed.

**Comment by lucatume** [comment]
Ok, changes are in, good for another (or a continuation of the previous) review.

**Pull request merged**